api/envoy/extensions/filters/http/fault/v3/fault.proto

```diff
@@ -148,4 +149,11 @@ message HTTPFault {
   // If set to false, dynamic stats storage will be allocated for the downstream cluster name.
   // Default value is false.
   bool disable_downstream_cluster_stats = 15;
+
+  // When an abort or delay fault is executed, the metadata provided here will be added to the
+  // request's dynamic metadata. The metadata must be set under the filter namespace that will use it.
```

Comment:
Why is it dynamic metadata here but filter state in https://github.com/envoyproxy/envoy/pull/27003?

Reply:
The dynamic metadata here is just purely protobuf struct. There is no stat or anything. This is equivalent to the per route metadata and cluster metadata. Its not conveying anything that is computed by this filter